### PR TITLE
April Release Notes

### PR DESCRIPTION
**Pull request by NWeber-SL**
Please review the release notes for your area:

- Fabs. New Ocular Tutorial
- Davy.  SOC 2 Type 1 Compliance
- Leandro. Inspect for .Net Improvements
- Nacho. Protect for Java Stability Improvements
- Markus. Inspect JSP Support

### CODE REVIEW

File: release-notes/release-notes.md [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
-None
+* [April 2019](#april-2019-release-notes)
+* [March 2019](#march-2019-release-notes)
+
+
+## April 2019 Release Notes
+
+### ShiftLeft Ocular
+
+* **New Tutorial on Creating a Code Property Graph (CPG) from a Java Archive**. The ShiftLeft Ocular documentation now includes a tutorial on creating a CPG from a Java Archive. ShiftLeft Ocular creates a CPG for only the application code, and with references to the dependency code. This distinction is made through the use of a built-in Smart Jar Unpacker, which heuristically determines for each namespace whether it contains application or dependency code.
```
Comment (vlad-alexandru-ionescu) [inline release-notes/release-notes.md]: We should include a link to the actual tutorial.

File: release-notes/release-notes.md [hunk 2]
```diff
@@ -1,9 +1,32 @@
...
+
+### ShiftLeft Inspect and ShiftLeft Protect
+
+* **SOC 2 Type 1 Compliance for ShiftLeft Inspect and ShiftLeft Protect**. ShiftLeft Inspect now analyzes, and ShiftLeft Protect secures, financial data through SOC 2 compliance. By providing specific visibility into how you are handling financial data, your customers' financial statements and financial reporting can now be protected against the exploitation of vulnerabilities. The Type 1 report assesses the design of security processes at a specific point in time.
```
Comment (vlad-alexandru-ionescu) [inline release-notes/release-notes.md]: We should hold-off on merging this into master until this is actually finalized. Davy will be able to keep us in the loop with that.

File: release-notes/release-notes.md [hunk 3]
```diff
@@ -1,9 +1,32 @@
...
+
+* **Enhancements to ShiftLeft Inspect for .Net**. The current version of ShiftLeft Inspect for .Net includes significant performance improvements in both processing time and memory consumption. In addition, numerous bug fixes have been made.
+
+* **JSP support added for ShiftLeft Inspect**. You can now use ShiftLeft Inspect to analyze your JSP pages for vulnerabilities and data leakage. Note that this support does not include JSP Expression Language.
+
+* **Stability Improvements to ShiftLeft Protect**. Stability improvements have been made in the ShiftLeft Protect Microagent.
```
Comment (vlad-alexandru-ionescu) [inline release-notes/release-notes.md]:
```suggestion
* **Stability Improvements to ShiftLeft Protect**. Stability improvements have been made in the ShiftLeft Protect Microagent for Java.
```

File: release-notes/release-notes.md (hunk 1 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Yes, you are right. Thanks!

File: release-notes/release-notes.md [hunk 4]
```diff
@@ -1,9 +1,34 @@
...
+* [March 2019](#march-2019-release-notes)
+
+
+## April 2019 Release Notes
+
+* **Update to ShiftLeft's Terms of Service**. The ShiftLeft [Terms of Service](https://www.shiftleft.io/terms/) have been updated for post-termination obligations (4d). 
+
+### ShiftLeft Ocular
+
+* **New Tutorial on Creating a Code Property Graph (CPG) from a Java Archive**. The ShiftLeft Ocular documentation now includes a [tutorial on creating a CPG from a Java Archive](../using-ocular/tutorials/cpgcreate.md). ShiftLeft Ocular creates a CPG for only the application code, and with references to the dependency code. This distinction is made through the use of a built-in Smart Jar Unpacker, which heuristically determines for each namespace whether it contains application or dependency code.
```
Comment (arunmecheri) [inline release-notes/release-notes.md]:
```suggestion
* **New Tutorial on Creating a Code Property Graph (CPG) from a Java Archive (JAR)**. The ShiftLeft Ocular documentation now includes a [tutorial on creating a CPG from a JAR](../using-ocular/tutorials/cpgcreate.md). ShiftLeft Ocular creates a CPG for only the application code, and with references to the dependency code. This distinction is made through the use of a built-in Smart Jar Unpacker, which heuristically determines for each namespace whether it contains application or dependency code.
```

File: release-notes/release-notes.md (hunk 1 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Done

File: release-notes/release-notes.md (hunk 2 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: OK, material removed from the release notes until further notice.

File: release-notes/release-notes.md (hunk 3 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Done

File: release-notes/release-notes.md (hunk 4 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Done

File: release-notes/release-notes.md [hunk 5]
```diff
@@ -1,9 +1,85 @@
...
+* **Update to ShiftLeft's Terms of Service**. The ShiftLeft [Terms of Service](https://www.shiftleft.io/terms/) have been updated for post-termination obligations (4d). 
+
+### ShiftLeft Ocular
+
+* **New Tutorial on Creating a Code Property Graph (CPG) from a Java Archive (JAR)**. The ShiftLeft Ocular documentation now includes a [tutorial on creating a CPG from a JAR](../using-ocular/tutorials/cpgcreate.md). ShiftLeft Ocular creates a CPG for only the application code, with references to the dependency code. This distinction is made through the use of a built-in Smart Jar Unpacker, which heuristically determines for each namespace whether it contains application or dependency code.
+
+#### ShiftLeft Ocular v0.3.23
+
+* **New `Config` Objects for Specifying Custom Policy and Other Configurations**. The following new `config` objects have been introduced in ShiftLeft Ocular for specifying custom policy and other configurations:
+  * `config.policy.dynamicPolicyPath("/path/to/dynamic/policy")`
```
Comment (tuxology) [inline release-notes/release-notes.md]: `ocular> config.policy.dynamicPolicyPath("/path/to/dynamic/policy")`

File: release-notes/release-notes.md [hunk 6]
```diff
@@ -1,9 +1,85 @@
...
+  * `config.policy.staticPolicyPath("/path/to/dynamic/policy")`
```
Comment (tuxology) [inline release-notes/release-notes.md]: `ocular> config.policy.staticPolicyPath("/path/to/static/policy")`

File: release-notes/release-notes.md [hunk 7]
```diff
@@ -1,9 +1,85 @@
...
+
+* **New `version` Command Added**. This new `version` command prints the current version string.
+
+* **Determining Aliases**. ShiftLeft Ocular has been enhanced to allow determining aliases for a list of identifiers in C/C++ code.
+
+#### ShiftLeft Ocular v0.3.20
+
+* **On-Disk Graph Swapping for Low RAM Machines.** This experimental feature allows CPG elements to be swapped on disk as required from the main memory. Note that this feature adds extra latency of loading/unloading of memory and is not yet officially supported. Use the following function on the ShiftLeft Ocular shell with either default or custom parameters to enable On-Disk Graph Swapping for Low RAM Machines: 
+  * `enableOnDiskOverflow()`
+  * `config.policy.staticPolicyPath("/path/to/dynamic/policy")`
```
Comment (tuxology) [inline release-notes/release-notes.md]: Remove this line

File: release-notes/release-notes.md [hunk 8]
```diff
@@ -1,9 +1,85 @@
...
+* **The `workspaceReset` Command Replaced with `workspace.reset`**. The `workspaceReset` command has been deprecated and replaced with the `workspace.reset` command. The new command resets the workspace and deletes all CPGs stored on disk.
+
+* **New `version` Command Added**. This new `version` command prints the current version string.
+
+* **Determining Aliases**. ShiftLeft Ocular has been enhanced to allow determining aliases for a list of identifiers in C/C++ code.
+
+#### ShiftLeft Ocular v0.3.20
+
+* **On-Disk Graph Swapping for Low RAM Machines.** This experimental feature allows CPG elements to be swapped on disk as required from the main memory. Note that this feature adds extra latency of loading/unloading of memory and is not yet officially supported. Use the following function on the ShiftLeft Ocular shell with either default or custom parameters to enable On-Disk Graph Swapping for Low RAM Machines: 
+  * `enableOnDiskOverflow()`
```
Comment (tuxology) [inline release-notes/release-notes.md]: `ocular> enableOnDiskOverflow()`

File: release-notes/release-notes.md [hunk 9]
```diff
@@ -1,9 +1,85 @@
...
+  * `config.policy.staticPolicyPath("/path/to/dynamic/policy")`
+
+* **Overlay API**. The Security Profile is now part of the CPG, as an overlay. This new feature unifies the Ocular Query Language (OQL) for the CPG and Security Profile and removes the need for using `cpg2sp.sh` to create a Security Profile. This means that all Security Profile functionality is now part of the CPG. For example, you now use `cpg.finding.p` instead of `sp.findings.p`. The new Overlay feature requires that `cpg2sp` to be run with the --overlay flag. For ease of use, all Overlay API endpoints have been integrated into ShiftLeft Ocular.
+
+* **Integrated CPG and SP Generation**. CPG and Security Profile generation can now be performed from inside ShiftLeft Ocular, with both CPGs and their overlays managed in a workspace. This new feature allows you to effectively work with multiple CPGs at once.
+  * `createCpg("/path/to/jar")`
+  * `createCpgAndSp("/path/to/jar")`
+  * `createSp(cpg)`
```
Comment (tuxology) [inline release-notes/release-notes.md]: For above 3 lines lets prefix  `ocular>` Just like above comments

File: release-notes/release-notes.md [hunk 10]
```diff
@@ -1,9 +1,85 @@
...
+
+ Older APIs and functionality (e.g. `loadCpg` and `loadSp`) are now backwards compatible.
+
+* **Workspaces**. ShiftLeft Ocular now includes workspaces for easy management of CPGs and overlays. [Refer to the API] (https://ocular.shiftleft.io/api/io/shiftleft/repl/Console.html) for additional information.
+
+* **Load Multiple CPG Queries**. You can now load more than one CPG in a given workspace and then combine queries. For example, `cpgs.flatMap(_.method.fullName.l)`.
```
Comment (tuxology) [inline release-notes/release-notes.md]: `ocular> cpgs.flatMap(_.method.fullName.l)`

File: release-notes/release-notes.md [hunk 11]
```diff
@@ -1,9 +1,85 @@
...
+
+* **Deprecated `sp` Object**. Functionality of the deprecated `sp` object has been transferred to `cpg` object.
+
+* **Dependency List Outputs Complete Dependencies**. The Dependency List has been fixed to now output complete dependencies.
+
+* **Renamed newLocation Step**. The newLocation step has been rename to location.
+
+* ** `.flows` Behavior Fixed**. `.flows` behavior has been fixed to show only a single flow, thereby resolving printing issues. Use `.allFlows` to show the complete flows list.
```
Comment (tuxology) [inline release-notes/release-notes.md]: behavior fixed

File: release-notes/release-notes.md [hunk 12]
```diff
@@ -1,9 +1,85 @@
...
+
+* **Workspaces**. ShiftLeft Ocular now includes workspaces for easy management of CPGs and overlays. [Refer to the API] (https://ocular.shiftleft.io/api/io/shiftleft/repl/Console.html) for additional information.
+
+* **Load Multiple CPG Queries**. You can now load more than one CPG in a given workspace and then combine queries. For example, `cpgs.flatMap(_.method.fullName.l)`.
+
+* **Deprecated `sp` Object**. Functionality of the deprecated `sp` object has been transferred to `cpg` object.
+
+* **Dependency List Outputs Complete Dependencies**. The Dependency List has been fixed to now output complete dependencies.
+
+* **Renamed newLocation Step**. The newLocation step has been rename to location.
```
Comment (tuxology) [inline release-notes/release-notes.md]: **Renamed `newLocation` Step**. The `newLocation` step has been rename to location.

File: release-notes/release-notes.md (hunk 5 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Done

File: release-notes/release-notes.md (hunk 6 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Done

File: release-notes/release-notes.md (hunk 7 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: The whole section on "On-Disk Graph Swapping"?

Comment (NWeber-SL) [reply release-notes/release-notes.md]: Removed `config.policy.staticPolicyPath("/path/to/dynamic/policy")`

File: release-notes/release-notes.md (hunk 8 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Done

File: release-notes/release-notes.md (hunk 9 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Done

File: release-notes/release-notes.md (hunk 10 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Done

File: release-notes/release-notes.md (hunk 11 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Don't understand what you want here.

File: release-notes/release-notes.md (hunk 12 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Don't understand what you want done here

Comment (NWeber-SL) [reply release-notes/release-notes.md]: Fixed

File: release-notes/release-notes.md (hunk 11 above)
Comment (NWeber-SL) [reply release-notes/release-notes.md]: Done

**Pull request merged**